File: Python/234-palindrome-linked-list/index2.py
# Definition for singly-linked list.
# class ListNode:
#   def __init__(self, x):
#     self.val = x
#     self.next = None

class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
      arr = []
      while head is not None:
        arr.append(head.val)
        head = head.next
      
      return arr == arr[::-1]

File: JavaScript/92-reverse-linked-list-ii/index2.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *   this.val = val;
 *   this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} m
 * @param {number} n
 * @return {ListNode}
 */
const reverseBetween = function (head, m, n) {
  if (!head) return null;

  let left = head;
  let right = head;
  let stop = false;

  const recursiveAndReverse = (right, m, n) => {
    if (n === 1) return;

    right = right.next;
    if (m > 1) left = left.next;
    recursiveAndReverse(right, m - 1, n - 1);

    if (left === right || right.next === left) {
      stop = true
    }

    if (!stop) {
      const temp = left.val;
      left.val = right.val;
      right.val = temp;

      left = left.next;
    }
  };

  recursiveAndReverse(right, m, n);
  return head;
};
